src/python/pants/backend/adhoc/target_types.py

```diff
@@ -253,6 +253,22 @@ class AdhocToolOutputRootDirField(StringField):
     )
 
 
+class AdhocToolHashOnlySourcesGlobsField(StringSequenceField):
+    alias: ClassVar[str] = "hash_only_sources_globs"
+    help = help_text(
+        """
+        Path globs for source files on which this target depends indirectly, but which should not be
+        materlized into the execution sandbox. Pants will compute the hash of all of the files
+        references by the globs and include that hash as part of the cache key for the
+        process to be executed (as an environment variable).
```

Comment:
I think it'd be good to be explicit about the consequence/purpose of this, e.g. something like:

```suggestion
        process to be executed (as an environment variable), so that the command re-runs if these files change.
```